### ISSUE TITLE
New exception in DC engine

### ISSUE THREAD

**Issue by baltzell** [JeffersonLab/clas12-offline-software#1017]
It's happening in multiple chefs' logs:

```
java.lang.ArrayIndexOutOfBoundsException: Index 3 out of bounds for length 3
    at org.jlab.service.dc.DCTBEngine.processDataEvent(DCTBEngine.java:202)
    at org.jlab.clas.reco.ReconstructionEngine.execute(ReconstructionEngine.java:376)
    at org.jlab.clara.sys.ServiceEngine.executeEngine(ServiceEngine.java:227)
    at org.jlab.clara.sys.ServiceEngine.execute(ServiceEngine.java:153)
    at org.jlab.clara.sys.Service.lambda$execute$2(Service.java:178)
    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.base/java.lang.Thread.run(Thread.java:834)
```

So, at line 202 here:

https://github.com/JeffersonLab/clas12-offline-software/blob/824f31685794aedf6e1a8b433c8cc7f2e29cecfc/reconstruction/dc/src/main/java/org/jlab/service/dc/DCTBEngine.java#L200-L206

**Comment by baltzell** [comment]
Has this been addressed?

**Comment by raffaelladevita** [comment]
Addressed by #1041
